### ISSUE TITLE
Crash When Computing Maxflow with Identical Source and Sink

### ISSUE THREAD

**Issue by iany0** [igraph/igraph#2437]
Hello,
**Describe the bug**  
A kernel crash occurs when attempting to compute the maximum flow (`maxflow`) in an igraph `Graph` object with the same node as both the source and the sink. The expected behavior would be to handle this scenario gracefully, either by returning a valid result (such as zero flow) or by raising an appropriate error message. However, the current behavior results in a kernel crash.

**To Reproduce**  
```python
g = ig.Graph()

g.add_vertices(7) 
g.add_edges([(0, 1), (0, 2), (1, 3), (2, 3), (2, 4), (3, 5), (4, 5), (6, 6)])
g.es['capacity'] = [1, 2, 3, 4, 5, 6, 7, 8]

print(g.maxflow(1, 1, capacity='capacity'))
```

**Expected Behavior**  
It would be preferable for the function to either return a valid result (such as a max flow of zero) or to appropriately raise an error when the source and sink are the same node. Ideally, a kernel crash should be avoided in these scenarios. This functionality works correctly in directed graphs.

**Version Information**  
`python-igraph` version: 0.11.2

**Comment by szhorvat** [comment]
Small example in C:


```
    igraph_t g;
    igraph_real_t flow_value;
    igraph_vector_t flow;

    igraph_vector_init(&flow, 0);

    igraph_small(&g, 0, IGRAPH_UNDIRECTED, 0, 1, 0, 2, 1, 2, -1);
    // igraph_to_directed(&g, IGRAPH_TO_DIRECTED_MUTUAL);
    igraph_maxflow(&g, &flow_value, &flow, NULL, NULL, NULL, 1, 1, NULL, NULL);
    igraph_vector_destroy(&flow);

    igraph_destroy(&g);
```

This program never finishes on my system.

If I pass `NULL` for `flow`, then it works fine. Directed or undirected does not matter.

**Comment by szhorvat** [comment]
It seems to go into an infinite loop in this loop:

https://github.com/igraph/igraph/blob/master/src/flow/flow.c#L995

**Comment by szhorvat** [comment]
@ntamas Do you have time to look at this?

**Comment by ntamas** [comment]
I'll take a look this afternoon.

**Comment by ntamas** [comment]
FWIW I would simply rule out using the same vertex for the source and the target of the flow by throwing an error. The argument for throwing an error is that the max-flow problem is the dual problem of the min-cut problem; saturated edges on a maximum flow graph yield a minimal cut between source and target. Since cuts are not defined for the same source-target vertex pair, flows should not be defined either to maintain duality. Otherwise we would claim that removing all the edges participating in the loops would separate the source from the target.

Also, it does not matter whether there are loop edges or not. The value of a flow is defined as the flow on the edges leaving the source minus the flow of the edges entering the source. No matter whether there are loops or not, the flow will be zero.

**Comment by szhorvat** [comment]
So just so this is written down: this is "fixed" in 0.10.8 in the sense that using the same source and target are forbidden. This prevents situations where people accidentally lock up their session.

I'd still like to think this through some more, so let's please leave this open for a while.
